File: app/src/main/java/com/kontranik/koreader/parser/fb2reader/model/FB2Elements.java
package com.kontranik.koreader.parser.fb2reader.model;

public enum  FB2Elements {
    a("a"),
    annotation("annotation"),
    author("author"),
    binary("binary"),
    body("body"),
    bookname("book-name"),
    booktitle("book-title"),
    cite("cite"),
    city("city"),
    code("code"),
    coverpage("coverpage"),
    custominfo("custom-info"),
    date("date"),
    description("description"),
    documentinfo("document-info"),
    email("email"),
    emphasis("emphasis"),
    emptyline("empty-line"),
    epigraph("epigraph"),
    fictionbook("FictionBook"),
    firstname("first-name"),
    genre("genre"),
    history("history"),
    homepage("home-page"),
    id("id"),
    image("image"),
    isbn("isbn"),
    keywords("keywords"),
    lang("lang"),
    lastname("last-name"),
    middlename("middle-name"),
    nickname("nickname"),
    p("p"),
    poem("poem"),
    programused("program-used"),
    publisher("publisher"),
    publishinfo("publish-info"),
    section("section"),
    sequence("sequence"),
    srclang("src-lang"),
    srcocr("src-ocr"),
    srcurl("src-url"),
    stanza("stanza"),
    strikethrough("strikethrough"),
    strong("strong"),
    style("style"),
    stylesheet("stylesheet"),
    sub("sub"),
    subtitle("subtitle"),
    sup("sup"),
    textauthor("text-author"),
    title("title"),
    titleinfo("title-info"),
    translator("translator"),
    v("v"),
    version("version"),
    year("year");

    public String elName;
    FB2Elements(String name) {
        this.elName = name;
    }

    public static FB2Elements fromString(String name) {
        for (FB2Elements b : FB2Elements.values()) {
            if (b.elName.equalsIgnoreCase(name)) {
                return b;
            }
        }
        return null;
    }
}

File: app/src/main/java/com/kontranik/koreader/parser/fb2reader/model/FB2Elements.kt
package com.kontranik.koreader.parser.fb2reader.model

enum class FB2Elements(var elName: String) {
    A("a"),
    ANNOTATION("annotation"),
    AUTHOR("author"),
    BINARY("binary"),
    BODY("body"),
    BOOKNAME("book-name"),
    BOOKTITLE("book-title"),
    CITE("cite"), city("city"),
    CODE("code"),
    COVERPAGE("coverpage"),
    CUSTOMINFO("custom-info"),
    DATE("date"), description("description"),
    DOCUMENTINFO("document-info"),
    EMAIL("email"),
    EMPHASIS("emphasis"),
    EMPTYLINE("empty-line"),
    EPIGRAPH("epigraph"),
    FICTIONBOOK("FictionBook"),
    FIRSTNAME("first-name"),
    GENRE("genre"),
    HISTORY("history"),
    HOMEPAGE("home-page"),
    ID("id"),
    IMAGE("image"),
    ISBN("isbn"),
    KEYWORDS("keywords"),
    LANG("lang"),
    LASTNAME("last-name"),
    MIDDLENAME("middle-name"),
    NICKNAME("nickname"),
    P("p"), poem("poem"),
    PROGRAMUSED("program-used"),
    PUBLISHER("publisher"),
    PUBLISHINFO("publish-info"),
    SECTION("section"),
    SEQUENCE("sequence"),
    SRCLANG("src-lang"),
    SRCOCR("src-ocr"),
    SRCURL("src-url"),
    STANZA("stanza"),
    STRIKETHROUGH("strikethrough"),
    STRONG("strong"),
    STYLE("style"),
    STYLESHEET("stylesheet"),
    SUB("sub"),
    SUBTITLE("subtitle"),
    SUP("sup"),
    TEXTAUTHOR("text-author"),
    TITLE("title"),
    TITLEINFO("title-info"),
    TRANSLATOR("translator"),
    V("v"),
    VERSION("version"),
    YEAR("year");

    companion object {
        fun fromString(name: String?): FB2Elements? {
            for (b in values()) {
                if (b.elName.equals(name, ignoreCase = true)) {
                    return b
                }
            }
            return null
        }
    }
}
